lib/constructs/persistence/data-pipeline.ts

```diff
@@ -202,4 +202,54 @@ export class SyllabusSyncPipeline extends AbstractDataPipeline {
             filters: [{prefix: 'syllabus/'}],
         }))
     }
-}
\ No newline at end of file
+}
+
+// update blog on notification
+export class BlogContentPipeline extends AbstractDataPipeline {
+
+    readonly dataSource: Bucket;
+
+    readonly processor: Function;
+
+    readonly dataWarehouse: Table;
+
+    constructor(scope: cdk.Construct, id: string, props?: DataPipelineProps) {
+        super(scope, id);
+
+        this.dataWarehouse = new Table(this, 'blog-table', {
+            partitionKey: {name: "title", type: AttributeType.STRING},
+            sortKey: {name: "ctime", type: AttributeType.STRING},
+            billingMode: BillingMode.PROVISIONED,
+            encryption: TableEncryption.DEFAULT,
+            removalPolicy: cdk.RemovalPolicy.RETAIN,
+            timeToLiveAttribute: "ttl",
+            tableName: "wasedatime-blog",
+            readCapacity: 1,
+            writeCapacity: 1,
+        });
+
+        this.dataSource = new Bucket(this, 'blog-bucket', {
+            accessControl: BucketAccessControl.PRIVATE,
+            blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
+            bucketName: "wasedatime-blog",
```

Comment:
The bucket should be public since we want assets inside to be directly accessed by the user.

Reply:
Have setted it to public read